minor-mode.lisp

```diff
@@ -0,0 +1,545 @@
+(in-package :stumpwm)
+
+(export '(define-minor-mode
+          minor-mode-id
+          minor-mode-scope
+          enable-minor-mode
+          disable-minor-mode
+          minor-mode-on-enable
+          minor-mode-on-disable
+          minor-mode-keymap
+          minor-mode-lighter
+          generate-keymap))
+
+(defvar *minor-mode*)
+
+(defvar *active-global-minor-modes* nil
```

Comment:
minor style suggestion, write nil as () here as it is used as a list instead of a boolean
```suggestion
(defvar *active-global-minor-modes* ()
```